internal/service/sesv2/dedicated_ip_assignment.go

```diff
@@ -0,0 +1,172 @@
+package sesv2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sesv2"
+	"github.com/aws/aws-sdk-go-v2/service/sesv2/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func ResourceDedicatedIPAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceDedicatedIPAssignmentCreate,
+		ReadWithoutTimeout:   resourceDedicatedIPAssignmentRead,
+		DeleteWithoutTimeout: resourceDedicatedIPAssignmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"destination_pool_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
Maybe [`validation.IsIPAddress`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/validation?utm_source=gopls#IsIPAddress) could be useful here.
```suggestion
			"ip": {
				Type:     schema.TypeString,
				Required: true,
				ForceNew: true,
				ValidateFunc: validation.IsIPAddress,
			},
```

Reply:
Good call, added 👍.